examples/serial_echo_rtic.rs

```diff
@@ -0,0 +1,106 @@
+#![no_main]
+#![no_std]
+
+use panic_rtt_target as _;
+
+#[rtic::app(device = stm32f3xx_hal::pac, dispatchers = [TIM20_BRK, TIM20_UP, TIM20_TRG_COM])]
+mod app {
+    use dwt_systick_monotonic::DwtSystick;
+    use rtt_target::{rprintln, rtt_init_print};
+    use stm32f3xx_hal::{prelude::*, serial::{Event, Serial}};
+    use stm32f3xx_hal::gpio::{self, Output, PushPull, Alternate, U};
+
+    #[monotonic(binds = SysTick, default = true)]
+    type DwtMono = DwtSystick<48_000_000>;
+
+    type SerialType = Serial<stm32f3xx_hal::pac::USART1, (gpio::Pin<gpio::Gpioa, U<9_u8>, Alternate<PushPull, 7_u8>>, gpio::Pin<gpio::Gpioa, U<10_u8>, Alternate<PushPull, 7_u8>>)>;
+    type DirType = stm32f3xx_hal::gpio::Pin<gpio::Gpioe, U<13_u8>, Output<PushPull>>;
```

Comment:
Even though these are the correct types hinted by the compiler, there exists type aliases, which are easier to read. I'd recommend them: 

```suggestion
    type SerialType = Serial<stm32f3xx_hal::pac::USART1, (gpio::gpioa::PA9<A7<PushPull>>, gpio::gpioa::PA10<A7<PushPull>>)>;
    type DirType = stm32f3xx_hal::gpio::gpioe::PE13<Output<PushPull>>;
```